k8s/anthos/clusters.go

```diff
@@ -0,0 +1,59 @@
+package anthos
+
+import (
+	"context"
+
+	v1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// ClusterOps is an interface to perform k8s cluster operations
+type ClusterOps interface {
+	// ListCluster lists all kubernetes clusters
+	ListCluster(ctx context.Context, options metav1.ListOptions) (*v1beta1.ClusterList, error)
+	// GetCluster returns a cluster for the given name
+	GetCluster(ctx context.Context, name string, options metav1.GetOptions) (*v1beta1.Cluster, error)
+	// DescribeCluster gets the cluster status
+	DescribeCluster(ctx context.Context, name string, options metav1.GetOptions) (*v1beta1.ClusterStatus, error)
+}
+
+// ListCluster lists all kubernetes clusters
+func (c *Client) ListCluster(ctx context.Context, options metav1.ListOptions) (result *v1beta1.ClusterList, err error) {
+	if err := c.initClient(); err != nil {
+		return nil, err
+	}
+	result = &v1beta1.ClusterList{}
+	err = c.RESTClient().Get().
+		Resource("clusters").
+		VersionedParams(&options, scheme.ParameterCodec).
+		Do(ctx).
+		Into(result)
+
+	return
+}
+
+// GetCluster returns a cluster for the given name
+func (c *Client) GetCluster(ctx context.Context, name string, options metav1.GetOptions) (result *v1beta1.Cluster, err error) {
+	if err := c.initClient(); err != nil {
+		return nil, err
+	}
+	result = &v1beta1.Cluster{}
```

Comment:
Change this to `cluster` like you are using in the function below, result is not very intuitive for this var, let's be consistent
NOTE: Also should it be `:=`, since this is first time you defining this var?

Reply:
Incorporated !!